src/main/java/org/codehaus/plexus/archiver/zip/AbstractZipUnArchiver.java

```diff
@@ -44,6 +43,9 @@
 
     private String encoding = "UTF8";
 
+    @Nullable
+    private Long maxOutputSize;
```

Comment:
Would it make sense to use `long` with `-1` meaning unlimited in order to avoid NPEs?

Reply:
Would it makes sense to have it for other archives (such as tar)?